integration/container/cdi_test.go

```diff
@@ -0,0 +1,92 @@
+package container // import "github.com/docker/docker/integration/container"
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	containertypes "github.com/docker/docker/api/types/container"
+	dclient "github.com/docker/docker/client"
+	"github.com/docker/docker/integration/internal/container"
+	"github.com/docker/docker/pkg/stdcopy"
+	"github.com/docker/docker/testutil/daemon"
+	"github.com/docker/docker/testutil/fakecontext"
+	"gotest.tools/v3/assert"
+	is "gotest.tools/v3/assert/cmp"
+	"gotest.tools/v3/skip"
+)
+
+func TestCreateWithCDIDevices(t *testing.T) {
+	skip.If(t, testEnv.OSType != "linux", "CDI devices are only supported on Linux")
+	skip.If(t, !testEnv.DaemonInfo.ExperimentalBuild, "experimental build required for CDI devices")
+	skip.If(t, testEnv.IsRemoteDaemon, "cannot run daemon when remote daemon")
```

Comment:
It looks like these two cases are handled when setting `client`, since it looks like we can start an experimental-enabled daemon for the test if not using a remote daemon.

If this is not possible due to, e.g., fakecontext, then keep these skips, but remove the code that attempts to start an experimental daemon, i.e. lines 46-51.

Reply:
Thanks for the clarification. I meant when using CDI outside of the test context. The only requirement would be that the spec be readable by the daemon. That means that if there were a mechanism to create the spec file on a remote host where the daemon was running and have it readable there, we would expect CDI injection to work as expected.

From you comment I would say that it is out of scope for this particular test.